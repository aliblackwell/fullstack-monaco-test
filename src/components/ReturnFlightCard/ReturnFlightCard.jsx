import React from 'react';
import PropTypes from 'prop-types';
import BpkCard from 'bpk-component-card';
import STYLES from './ReturnFlightCard.scss';
import FlightCardDetail from '../FlightCardDetail';
const getClassName = (className) => STYLES[className] || 'UNKNOWN';

const ReturnFlightCard = ({ outboundFlight, inboundFlight, price, url }) => {
  return (
    <section>
      <BpkCard className={getClassName('ReturnFlightCard')}>
        <FlightCardDetail flight={outboundFlight} />
        <FlightCardDetail flight={inboundFlight} />
      </BpkCard>
    </section>
  );
};

ReturnFlightCard.propTypes = {
  outboundFlight: PropTypes.object,
  inboundFlight: PropTypes.object,
  price: PropTypes.string,
  url: PropTypes.string,
};

export default ReturnFlightCard;
