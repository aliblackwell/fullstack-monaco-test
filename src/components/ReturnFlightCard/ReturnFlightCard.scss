@import '~bpk-mixins';

.ReturnFlightCard {
  max-width: 480px;
  min-width: 260px;
  margin: 0px auto $bpk-spacing-md;
  &__layout-container {
    display: flex;
    justify-content: space-between;
  }
}
