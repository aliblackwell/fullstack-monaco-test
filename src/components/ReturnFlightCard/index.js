import ReturnFlightCard from "./ReturnFlightCard";

export default ReturnFlightCard;