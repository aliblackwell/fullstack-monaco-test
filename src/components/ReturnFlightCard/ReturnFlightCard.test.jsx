import React from 'react';
import TestRenderer from 'react-test-renderer';
import ReturnFlightCard from './ReturnFlightCard';

describe('ReturnFlightCard', () => {
  it('should render correctly', () => {
    const props = {
      outboundFlight: {
        id: 'leg_1',
        departure_airport: 'BUD',
        arrival_airport: 'LTN',
        departure_time: '2020-10-31T15:35',
        arrival_time: '2020-10-31T17:00',
        stops: 0,
        airline_name: 'Wizz Air',
        airline_id: 'WZ',
        duration_mins: 145,
      },
      inboundFlight: {
        id: 'leg_6',
        departure_airport: 'LGW',
        arrival_airport: 'BUD',
        departure_time: '2020-11-11T08:10',
        arrival_time: '2020-11-11T11:40',
        stops: 0,
        airline_name: 'Lufthansa',
        airline_id: 'LH',
        duration_mins: 150,
      },
      price: 'Â£150',
      agent: 'Some agent',
    };

    const tree = TestRenderer.create(<ReturnFlightCard {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
