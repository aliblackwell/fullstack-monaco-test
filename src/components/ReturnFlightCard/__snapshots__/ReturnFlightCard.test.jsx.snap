// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReturnFlightCard should render correctly 1`] = `
<section>
  <div
    className="bpk-card bpk-card--padded ReturnFlightCard"
    role="button"
    tabIndex={0}
  >
    <article
      className="FlightCardDetail"
    >
      <div
        className="FlightCardDetail__leg-container"
      >
        <div
          className="FlightCardDetail__left"
        >
          <div
            className={null}
            style={
              Object {
                "height": "32px",
                "minWidth": "32px",
                "width": "32px",
              }
            }
          >
            <div
              className="bpk-image bpk-image--no-background"
              style={
                Object {
                  "height": 0,
                  "paddingBottom": "100%",
                }
              }
              suppressHydrationWarning={false}
            >
              <img
                alt="Wizz Air logo"
                className="bpk-image__img"
                onLoad={[Function]}
                src="https://logos.skyscnr.com/images/airlines/favicon/WZ.png"
                suppressHydrationWarning={false}
              />
            </div>
          </div>
          <div
            className="FlightCardDetail__times-container"
          >
            <p
              className="bpk-text bpk-text--base"
            >
              15:35
              <br />
              <span
                className="bpk-text bpk-text--base FlightCardDetail__airport"
              >
                BUD
              </span>
            </p>
            <svg
              aria-label="Arrow"
              className="abc-icon__arrow"
              height="18"
              style={
                Object {
                  "height": "1.125rem",
                  "width": "1.125rem",
                }
              }
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M3 12a1.5 1.5 0 0 0 1.5 1.5h11.379l-4.94 4.94a1.5 1.5 0 0 0 2.122 2.12l7.5-7.5a1.5 1.5 0 0 0 0-2.12l-7.5-7.5a1.5 1.5 0 0 0-2.122 2.12l4.94 4.94H4.5A1.5 1.5 0 0 0 3 12z"
              />
            </svg>
            <p
              className="bpk-text bpk-text--base"
            >
              17:00
              <br />
              <span
                className="bpk-text bpk-text--base FlightCardDetail__airport"
              >
                LTN
              </span>
            </p>
          </div>
        </div>
        <div
          className="FlightCardDetail__right"
        >
          <p
            className="bpk-text bpk-text--sm FlightCardDetail__flight-meta"
          >
            2h 25
            <br />
            <span
              className="bpk-text bpk-text--sm FlightCardDetail__flight-stops_direct"
            >
              Direct
            </span>
          </p>
        </div>
      </div>
    </article>
    <article
      className="FlightCardDetail"
    >
      <div
        className="FlightCardDetail__leg-container"
      >
        <div
          className="FlightCardDetail__left"
        >
          <div
            className={null}
            style={
              Object {
                "height": "32px",
                "minWidth": "32px",
                "width": "32px",
              }
            }
          >
            <div
              className="bpk-image bpk-image--no-background"
              style={
                Object {
                  "height": 0,
                  "paddingBottom": "100%",
                }
              }
              suppressHydrationWarning={false}
            >
              <img
                alt="Lufthansa logo"
                className="bpk-image__img"
                onLoad={[Function]}
                src="https://logos.skyscnr.com/images/airlines/favicon/LH.png"
                suppressHydrationWarning={false}
              />
            </div>
          </div>
          <div
            className="FlightCardDetail__times-container"
          >
            <p
              className="bpk-text bpk-text--base"
            >
              08:10
              <br />
              <span
                className="bpk-text bpk-text--base FlightCardDetail__airport"
              >
                LGW
              </span>
            </p>
            <svg
              aria-label="Arrow"
              className="abc-icon__arrow"
              height="18"
              style={
                Object {
                  "height": "1.125rem",
                  "width": "1.125rem",
                }
              }
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M3 12a1.5 1.5 0 0 0 1.5 1.5h11.379l-4.94 4.94a1.5 1.5 0 0 0 2.122 2.12l7.5-7.5a1.5 1.5 0 0 0 0-2.12l-7.5-7.5a1.5 1.5 0 0 0-2.122 2.12l4.94 4.94H4.5A1.5 1.5 0 0 0 3 12z"
              />
            </svg>
            <p
              className="bpk-text bpk-text--base"
            >
              11:40
              <br />
              <span
                className="bpk-text bpk-text--base FlightCardDetail__airport"
              >
                BUD
              </span>
            </p>
          </div>
        </div>
        <div
          className="FlightCardDetail__right"
        >
          <p
            className="bpk-text bpk-text--sm FlightCardDetail__flight-meta"
          >
            2h 30
            <br />
            <span
              className="bpk-text bpk-text--sm FlightCardDetail__flight-stops_direct"
            >
              Direct
            </span>
          </p>
        </div>
      </div>
    </article>
    <div
      className="ReturnFlightCard__layout-container"
    >
      <div>
        <p
          className="bpk-text bpk-text--xl"
        >
          Â£150
        </p>
        <p
          className="bpk-text bpk-text--lg"
          style={
            Object {
              "color": "rgb(118,112,132)",
            }
          }
        >
          Some agent
        </p>
      </div>
      <button
        className="bpk-button ReturnFlightCard__button"
        disabled={false}
        onClick={null}
        type="button"
      >
        Select
      </button>
    </div>
  </div>
</section>
`;
